### PR TITLE
Missing bool tests

### PR DESCRIPTION
**Pull request by bramtayl**
[Uncovered](https://codecov.io/gh/JuliaLang/julia/src/master/base/bool.jl#L6)
[Uncovered](https://codecov.io/gh/JuliaLang/julia/src/master/base/bool.jl#L7)
[Uncovered](https://codecov.io/gh/JuliaLang/julia/src/master/base/bool.jl#L83)

### CODE REVIEW

**Review by dkarrasch — APPROVED** [review]

File: test/numbers.jl [hunk 1]
```diff
@@ -63,6 +63,11 @@ end
 
     @test iszero(false) && !iszero(true)
     @test isone(true) && !isone(false)
+
+    @test !typemin(Bool)
```
Comment (StefanKarpinski) [inline test/numbers.jl]:
```suggestion
    @test typemin(Bool) === false
```

File: test/numbers.jl [hunk 2]
```diff
@@ -63,6 +63,11 @@ end
...
+    @test typemax(Bool)
```
Comment (StefanKarpinski) [inline test/numbers.jl]:
```suggestion
    @test typemax(Bool) === true
```

File: test/numbers.jl [hunk 3]
```diff
@@ -63,6 +63,11 @@ end
...
+    @test !abs(false)
```
Comment (StefanKarpinski) [inline test/numbers.jl]:
```suggestion
    @test abs(false) === false
```

File: test/numbers.jl [hunk 4]
```diff
@@ -63,6 +63,11 @@ end
...
+    @test abs(true)
```
Comment (StefanKarpinski) [inline test/numbers.jl]:
```suggestion
    @test abs(true) === true
```

**Comment by StefanKarpinski** [comment]
Seems clearer to spell out which boolean value each test returns.

**Comment by bramtayl** [comment]
Ok that makes sense

**Pull request merged**